A14-channelz.md

```diff
@@ -0,0 +1,1051 @@
+SocketRefs.  Note that only the id is necessary to query channelz.
+
+The data representation of each ref:
+
+![message ref structure][message ref structure]
+
+
+## Channels and Subchannels
+
+Channels and Subchannels, or descendent channel, are hierarchically organized
+into a DAG structure.  The union of all channels and subchannels may not contain
+a cycle.  A descendent channel may have any number of descendent channels. Each
+descendent channel may also have any number of sockets.   However, a given 
+descendent channel cannot have heterogeneous children.  That is, a channel or
+subchannel may have descendent channels, or have sockets, but not both.
+
+A subchannel represents a channel that is load-balanced over.  When a channel
+has both subchannels and channels, the channels are not delegated to by the 
+superchannel.  An example would be a channel that is used by the ancestor 
+channel, but does not handle RPCs given by the application.
+
+Each channel has a ChannelRef reference which includes the channel id and an
+optional name.  The name is included for human consumption.  There are no
+restrictions on the name but it should be limited to a reasonable length.
+
+Each subchannel has a SubchannelRef reference which includes the subchannel id
+and an optional name.  The name is included for human consumption.  There are 
+no restrictions on the name but it should be limited to a reasonable length.
+
+![channel hierarchy 1][channel hierarchy 1]
+
+### Channel Data
+
+Channels and Subchannels include data about themselves, notably their initial
+parameters and their call stats.  Currently the data includes:
+
+* Channel state ( See: connectivity-semantics-and-api.md )
+* Channel trace
+* Channel target, if applicable.
+* Number of calls starts, succeeded, and failed.  Note this is slightly
+  different than the number of streams.  See the Socket Data section.
+* Time of the last call started.
+
+In general, each piece of data included is specific to the channel itself and
+NOT of its descendent channels.  This is to say that an ancestor channel
+is not an aggregator for descendent channels.  The target of a channel or
+subchannel may be different from that of its parent, or not present at all.  The
+channel state may be different from the ancestor channel state.   The number of
+calls started, succeeded, and failed should reflect if calls were specifically
+tied to the channel.
+
+The number of calls started, succeeded, and failed on a channel gives insight
+into the activity of the channel.  Subtracting the number of failed and
+succeeded calls from the number started indicates how many are in flight.  The
+total number of started calls indicates how active the channel has been over
+time.  The ratio between succeeded calls and failed calls gives a rough
+estimation of how healthy the channel has been.  The last call started timestamp
+indicates how close the channel is towards entering the idle state.
+
+#### Channel Trace
```

Comment:
@ncteisen, please review this section.

Are we intending that the channel trace design will be a separate doc?

Reply:
This section looks good to me as a stub. The design for channel tracing won't be that long. Once I do it, I can update this doc in a subsequent pull request. No need to block this for that.